bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataGrpcClient.java

```diff
@@ -238,31 +240,65 @@ public Row readModifyWriteRow(ReadModifyWriteRowRequest request) {
 
   @Override
   public ResultScanner<Row> readRows(ReadRowsRequest request) {
-    return readRows(request, retryOptions.enableRetries());
+    return readRows(request, retryOptions.enableRetries(), false);
+  }
+
+  @Override
+  public ResultScanner<Row> readRows(ReadRowsRequest request, boolean resumable) {
+    return readRows(request, retryOptions.enableRetries(), resumable);
   }
 
   /**
-   * Begin reading rows, optionally with a resumable scanner.
+   * Begin reading rows, optionally with a retriable or resumable scanner.
    */
-  private ResultScanner<Row> readRows(ReadRowsRequest request, boolean resumable) {
-    // Delegate all resumable operations to the scanner. It will request a non-resumable
-    // scanner during operation.
+  private ResultScanner<Row> readRows(
+      ReadRowsRequest request, boolean retriable, boolean resumable) {
+    if (retriable) {
+      return getRetriableScanner(request);
+    }
+    
     if (resumable) {
-      return new ResumingStreamingResultScanner(
-        retryOptions,
-          request,
-          new BigtableResultScannerFactory() {
-            @Override
-            public ResultScanner<Row> createScanner(ReadRowsRequest request) {
-              return readRows(request, false);
-            }
-          });
+      return getResumableScanner(request);
     }
 
-    final Call<ReadRowsRequest , ReadRowsResponse> readRowsCall =
+    return getNonresumableScanner(request);
+  }
+
+  /**
+   * Delegate all retriable operations to the scanner. It will request a non-retriable. scanner
+   * during operation. A reriable scanner is resumable.
+   */
+  private ResultScanner<Row> getRetriableScanner(ReadRowsRequest request) {
+    return new RetryingStreamingResultScanner(
+        retryOptions, request, new BigtableResultScannerFactory() {
+          @Override
+          public ResultScanner<Row> createScanner(ReadRowsRequest request) {
+            return readRows(request, false, false);
+          }
+        });
+  }
+
+  /**
+   * Returns a resumable scanner needed for rescan.
+   */
+  private ResultScanner<Row> getResumableScanner(ReadRowsRequest request) {
+    return  new RetryingStreamingResultScanner(
+      retryOptions, request, new BigtableResultScannerFactory() {
```

Comment:
Can we create a member variable for BigtableResultScannerFactory?  Can we change the implementation to call getNonresumableScanner() [or what ever other name we chose]?